tools/cwrap/plugins/WrapDim.py

```diff
@@ -0,0 +1,26 @@
+from . import CWrapPlugin
+from string import Template
+
+
+class WrapDim(CWrapPlugin):
+
+    DIM_WRAP_TEMPLATE = Template(
+        """if (${arg_to_wrap} < 0) ${arg_to_wrap} += ${arg_tensor}->nDimension;""")
```

Comment:
Actually it would be better to do some additional checks. Otherwise people might get some weird error messages, because the values might still be invalid even after shifting them. Just add
```c
THPUtils_assert(${arg_to_wrap} >= -nDimension && ${arg_to_wrap} < nDimension,
                "dimension out of range (expected to be in range of [%d, %d], but got %d)",
                ...)
```